tee-worker/app-libs/parentchain-interface/src/integritee/event_handler.rs

```diff
@@ -315,53 +315,37 @@ where
 
 		let extrinsic_sender = ParachainExtrinsicSender::new();
 
-		let update_token_staking_amount_call_index = self
+		let distribute_rewards_call_index = self
 			.node_metadata_repository
-			.get_from_metadata(|m| m.update_token_staking_amount_call_indexes())
+			.get_from_metadata(|m| m.distribute_rewards_call_indexes())
 			.map_err(|_| {
-				Error::Other(
-					"Metadata retrieval for update_token_staking_amount_call_indexes failed".into(),
-				)
+				Error::Other("Metadata retrieval for distribute_rewards_call_indexes failed".into())
 			})?
 			.map_err(|_| Error::Other("Invalid metadata".into()))?;
 
-		for account_id in account_ids.iter() {
-			let default_id_graph = Vec::new();
-			let id_graph = accounts_graphs.get(account_id).unwrap_or(&default_id_graph);
-			let staking_amount: Balance = id_graph
-				.iter()
-				.filter_map(|identity| {
-					let delegator = delegator_states.get(identity)?;
-					Some(delegator.total)
-				})
-				.sum();
-			let call = OpaqueCall::from_tuple(&(
-				update_token_staking_amount_call_index,
-				account_id,
-				staking_amount,
-				round_index,
-			));
-			extrinsic_sender
-				.send(call)
-				.map_err(|_| Error::Other("Failed to send extrinsic".into()))?;
-		}
-
-		let complete_reward_distribution_call_index = self
-			.node_metadata_repository
-			.get_from_metadata(|m| m.complete_reward_distribution_call_indexes())
-			.map_err(|_| {
-				Error::Other(
-					"Metadata retrieval for complete_reward_distribution_call_indexes failed"
-						.into(),
-				)
-			})?
-			.map_err(|_| Error::Other("Invalid metadata".into()))?;
-
-		let complete_reward_distribution_call =
-			OpaqueCall::from_tuple(&(complete_reward_distribution_call_index.clone()));
-		extrinsic_sender.send(complete_reward_distribution_call).map_err(|_| {
-			Error::Other("Failed to send complete_reward_distribution_call extrinsic".into())
-		})?;
+		let id_graphs_staking: Vec<(AccountId, Balance)> = account_ids
```

Comment:
I'm just concerned that the `account_ids` is unbounded - can we do it in batch?

Reply:
Yes, it makes sense. what do you think would be a sensible batch size? like 1000k ?